components/api_server/tests/routes/test_logo.py

```diff
@@ -1,16 +1,21 @@
 """Unit tests for the logo route."""
 
 import unittest
-from unittest.mock import patch
+from unittest.mock import Mock, patch
+
+import bottle
 
 from routes import get_logo
 
 
 class LogoTest(unittest.TestCase):
     """Unit tests for the logo route."""
 
-    @patch("bottle.static_file")
-    def test_logo(self, mock_static_file):
+    @patch("importlib.resources.read_binary", Mock(return_value="logo"))
+    def test_logo(self):
         """Test that a logo can be retrieved."""
-        mock_static_file.return_value = "logo"
         self.assertEqual("logo", get_logo("sonarqube"))
+
+    def test_missing_logo(self):
+        """Test that retrieving a missing logo results in a 404."""
+        self.assertRaises(bottle.HTTPError, get_logo, "missing")
```

Comment:
Don't we want to listen for `Logo not found: '{source_type}' is not a valid source type` here as well?

Reply:
No, because that error is checked when validating the data model and discovered when running unit test. The most likely scenario is adding a new source type but forgetting to add a logo for it. So that should never happen in production.